recipes-bsp/firmware-imx/firmware-nxp-wifi_1.0.bb

```diff
@@ -26,21 +27,97 @@ do_compile() {
     install -m 0644 cyw-wifi-bt/*_CYW*/BCM4349B1*.hcd ${D}${sysconfdir}/firmware
+
+    # Install NXP Connectivity common
+    install -d ${D}${nonarch_base_libdir}/firmware/nxp
+    install -m 0644 nxp/wifi_mod_para.conf ${D}${nonarch_base_libdir}/firmware/nxp
+
+    # Install NXP Connectivity SD8801 firmware
+    install -m 0644 nxp/FwImage_8801_SD/ed_mac_ctrl_V1_8801.conf  ${D}${nonarch_base_libdir}/firmware/nxp
+    install -m 0644 nxp/FwImage_8801_SD/sd8801_uapsta.bin         ${D}${nonarch_base_libdir}/firmware/nxp
+
+    # Install NXP Connectivity SDIO8987 firmware
+    install -m 0644 nxp/FwImage_8987/ed_mac_ctrl_V3_8987.conf  ${D}${nonarch_base_libdir}/firmware/nxp
+    install -m 0644 nxp/FwImage_8987/sdiouart8987_combo_v0.bin ${D}${nonarch_base_libdir}/firmware/nxp
+    install -m 0644 nxp/FwImage_8987/txpwrlimit_cfg_8987.conf  ${D}${nonarch_base_libdir}/firmware/nxp
+
+    # Install NXP Connectivity PCIE8997 firmware
+    install -m 0644 nxp/FwImage_8997/ed_mac_ctrl_V3_8997.conf  ${D}${nonarch_base_libdir}/firmware/nxp
+    install -m 0644 nxp/FwImage_8997/pcieuart8997_combo_v4.bin ${D}${nonarch_base_libdir}/firmware/nxp
+    install -m 0644 nxp/FwImage_8997/txpwrlimit_cfg_8997.conf  ${D}${nonarch_base_libdir}/firmware/nxp
+
+    # Install NXP Connectivity SDIO8997 firmware
+    install -m 0644 nxp/FwImage_8997_SD/ed_mac_ctrl_V3_8997.conf  ${D}${nonarch_base_libdir}/firmware/nxp
+    install -m 0644 nxp/FwImage_8997_SD/sdiouart8997_combo_v4.bin ${D}${nonarch_base_libdir}/firmware/nxp
+    install -m 0644 nxp/FwImage_8997_SD/txpwrlimit_cfg_8997.conf  ${D}${nonarch_base_libdir}/firmware/nxp
+
+    # Install NXP Connectivity PCIE9098 firmware
+    install -m 0644 nxp/FwImage_9098_PCIE/ed_mac_ctrl_V3_909x.conf  ${D}${nonarch_base_libdir}/firmware/nxp
+    install -m 0644 nxp/FwImage_9098_PCIE/pcieuart9098_combo_v1.bin ${D}${nonarch_base_libdir}/firmware/nxp
+    install -m 0644 nxp/FwImage_9098_PCIE/txpwrlimit_cfg_9098.conf  ${D}${nonarch_base_libdir}/firmware/nxp
+
+    # Install NXP Connectivity SDIO9098 firmware
+    install -m 0644 nxp/FwImage_9098_SD/sdiouart9098_combo_v1.bin ${D}${nonarch_base_libdir}/firmware/nxp
 }
 
 PACKAGES =+ " \
     ${PN}-bcm4359-pcie \
-"
+    ${PN}-nxp-common \
+    ${PN}-nxp8801-sdio \
+    ${PN}-nxp8987-sdio \
+    ${PN}-nxp8997-common \
+    ${PN}-nxp8997-pcie \
+    ${PN}-nxp8997-sdio \
+    ${PN}-nxp9098-pcie \
+    ${PN}-nxp9098-sdio"
 
 FILES:${PN}-bcm4359-pcie = " \
     ${nonarch_base_libdir}/firmware/brcm/brcmfmac4359-pcie.* \
-    ${sysconfdir}/firmware/BCM4349B1_*.hcd \
-"
-
+    ${sysconfdir}/firmware/BCM4349B1_*.hcd"
 RPROVIDES:${PN}-bcm4359-pcie = "linux-firmware-bcm4359-pcie"
+
+FILES:${PN}-nxp-common = " \
+    ${nonarch_base_libdir}/firmware/nxp/wifi_mod_para.conf"
+
+FILES:${PN}-nxp8801-sdio = " \
+    ${nonarch_base_libdir}/firmware/nxp/*8801*"
+RDEPENDS:${PN}-nxp8801-sdio  += "${PN}-nxp-common"
+RPROVIDES:${PN}-nxp8801-sdio += "linux-firmware-sd8801"
```

Comment:
Why are you adding this `RPROVIDES` set? It doesn't come from `linux-firmware`, and we should use the right package in depends on allowing easier find of what provides them. It also allows easier rework if those are included in `linux-firmware`.